src/Init/Core.lean

```diff
@@ -167,13 +167,14 @@ protected def bind {α : Type u} {β : Type v} (x : Task α) (f : α → Task β
 
 end Task
 
-/-- Some type that is not a scalar value in our runtime. -/
-structure NonScalar where
-  val : Nat
+private inductive PNonScalarSpec : Type u where
+  | mk (v : Nat)
 
 /-- Some type that is not a scalar value in our runtime and is universe polymorphic. -/
-inductive PNonScalar : Type u where
-  | mk (v : Nat) : PNonScalar
+opaque PNonScalar : Type u := PNonScalarSpec
```

Comment:
We should probably have a `Nonempty PNonScalar.{u}` instance.

Reply:
I wanted to see if we could avoid that, but the build errors are saying no. I've introduced a `Nonempty PNonScalar` as well as `unsafe def PNonScalar.null := unsafeCast (0 : Nat)` for use in the array functions.